pkg/controller/clusterset/clusterset_test.go

```diff
@@ -666,5 +671,127 @@ var _ = Describe("ClusterSet Controller", Label("controller"), Label("ClusterSet
 					Should(BeTrue())
 			})
 		})
+
+		When("created with ResourceQuota", func() {
+			It("should create resourceQuota if Quota is enabled", func() {
+				clusterSet := &v1alpha1.ClusterSet{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "clusterset-",
+						Namespace:    namespace,
+					},
+					Spec: v1alpha1.ClusterSetSpec{
+						Quota: &v1alpha1.Quota{
+							Limits: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("800m"),
+								v1.ResourceMemory: resource.MustParse("1Gi"),
+							},
+						},
+					},
+				}
+
+				err := k8sClient.Create(ctx, clusterSet)
+				Expect(err).To(Not(HaveOccurred()))
+				time.Sleep(3 * time.Second)
+				// wait for a bit for the ResourceQuota to be created
```

Comment:
The Eventually is used exactly for this, so I would remove the Sleep. It could also cause some flaky tests, what if the ResourceQuota takes more than 3s?

In the next Eventually I would just poll for the ResourceQuota. Once ready we can check for the values.